src/mlpack/methods/decision_tree/gini_gain.hpp

```diff
@@ -88,17 +89,23 @@ class GiniGain
 
       // SIMD loop: add counts for four elements simultaneously (if the compiler
       // manages to vectorize the loop).
+      // #pragma omp parallel for reduction (+:accWeights)
       for (size_t i = 3; i < labels.n_elem; i += 4)
       {
         const double weight1 = weights[i - 3];
         const double weight2 = weights[i - 2];
         const double weight3 = weights[i - 1];
         const double weight4 = weights[i];
 
+        // #pragma omp atomic
         counts[labels[i - 3]] += weight1;
+        // #pragma omp atomic
         counts2[labels[i - 2]] += weight2;
+        // #pragma omp atomic
         counts3[labels[i - 1]] += weight3;
+        // #pragma omp atomic
         counts4[labels[i]] += weight4;
+        // cannot add any counts in reduction as it shows some error.
```

Comment:
Could you paste the error here (in comments)?

Reply:
Ohh, count  must be scalar, never mind.

Reply:
Yes count needs to be a scalar.